src/chpasswd.c

```diff
@@ -432,6 +432,15 @@ int main (int argc, char **argv)
 			*cp = '\0';
 		} else {
 			if (feof (stdin) == 0) {
+
+				// Drop all remaining characters on this line.
+				while (fgets (buf, (int) sizeof buf, stdin) != (char *) 0) {
+					cp = strchr (buf, '\n');
+					if (cp != NULL) {
+						break;
```

Comment:
Would it be safer to just exit here?

Reply:
There are a few resources that are cleaned up and messages that are written to the screen after the outer loop finishes.